### PR TITLE
Frontend: Demo in madlib

### PR DESCRIPTION
**Pull request by benhammondmusic**
# Description and Motivation

- specify "at birth" on sex demographic type, capitalize `/Ethnicity`
- adds demographic selector component inside MadLib
- new font size that's fluid based on window width, with clamped min/max size
- defaults to race/eth if available, otherwise the first valid demo option for the given topics
- better semantic html for some components, fix some biome caught a11y issues
- simplify e2e tests; update with new demo location and values

## Has this been tested? How?

- tests updated and passing

## Screenshots (if appropriate)

Dynamic text size:

https://github.com/user-attachments/assets/1be08d2d-21c2-4130-b81d-506f5cb84d52

Screenshot 2024-12-12 at 9 11 08 PM
Screenshot 2024-12-12 at 9 11 39 PM
Screenshot 2024-12-12 at 9 13 08 PM

## Types of changes

(leave all that apply)

- New content or feature

### CODE REVIEW

File: frontend/playwright-tests/chronic_kidney_disease.nightly.spec.ts [hunk 1]
```diff
@@ -15,7 +15,6 @@ test('Chronic Kidney Disease', async ({ page }) => {
   await page.getByRole('dialog').getByText('Measurement Definition:').click()
   await page.getByText('For specific calculations and')
   await page.getByLabel('close topic info modal').click()
-  await page.getByText('Demographic').nth(2).click()
   await page.getByText('Off').nth(1).click()
   await page.locator('#menu- div').first().click()
```
Comment (kccrtv) [inline frontend/playwright-tests/chronic_kidney_disease.nightly.spec.ts:19]: we don't seem to need this locator line anymore

File: frontend/playwright-tests/ccb_adherence.nightly.spec.ts [hunk 2]
```diff
@@ -4,15 +4,16 @@ test('calcium channel blocker adherence', async ({ page }) => {
...
   await page.locator('.MuiBackdrop-root').click()
   await page
     .locator('#rate-map')
     .getByRole('heading', { name: 'Population adherent to' })
     .click()
   await page.getByLabel('open the topic info modal').click()
   await page.getByLabel('close topic info modal').click()
-  await page.getByText('Demographic', { exact: true }).nth(2).click()
   await page.getByText('Off').nth(1).click()
   await page.locator('#menu- div').first().click()
```
Comment (kccrtv) [inline frontend/playwright-tests/ccb_adherence.nightly.spec.ts:18]: rm

File: frontend/playwright-tests/care_avoidance_due_cost.nightly.spec.ts [hunk 3]
```diff
@@ -2,9 +2,11 @@ import { test } from '@playwright/test'
...
   await page.goto('/exploredata?mls=1.avoided_care-3.00&group1=All')
-  await page.getByText('Race and Ethnicity:').click()
+  await page
+    .getByRole('button', { name: 'Race and Ethnicity', exact: true })
+    .click()
   await page.locator('.MuiBackdrop-root').click()
-  await page.getByText('Demographic').nth(2).click()
+
   await page.getByText('Off').nth(1).click()
   await page.locator('#menu- div').first().click()
```
Comment (kccrtv) [inline frontend/playwright-tests/care_avoidance_due_cost.nightly.spec.ts:11]: rm

File: frontend/playwright-tests/beta_blockers.nightly.spec.ts [hunk 4]
```diff
@@ -4,15 +4,12 @@ test('Beta Blockers Adherence', async ({ page }) => {
...
-  await page.locator('.MuiBackdrop-root').click()
   await page
     .locator('#rate-map')
     .getByRole('heading', {
       name: 'Population adherent to beta blockers in the United States',
     })
     .click()
-  await page.getByText('Demographic', { exact: true }).nth(2).click()
   await page.getByText('Off').nth(1).click()
   await page.locator('#menu- div').first().click()
```
Comment (kccrtv) [inline frontend/playwright-tests/beta_blockers.nightly.spec.ts:14]: rm

File: frontend/playwright-tests/asthma.ci.spec.ts [hunk 5]
```diff
@@ -10,7 +10,6 @@ test('Asthma', async ({ page }) => {
     .click()
   await page.getByLabel('open the topic info modal').click()
   await page.getByLabel('close topic info modal').click()
-  await page.getByText('Demographic').nth(2).click()
   await page.getByText('Off').nth(1).click()
   await page.locator('#menu- div').first().click()
```
Comment (kccrtv) [inline frontend/playwright-tests/asthma.ci.spec.ts:14]: rm

File: frontend/playwright-tests/chronic_kidney_disease.nightly.spec.ts (hunk 1 above)
Comment (benhammondmusic) [reply frontend/playwright-tests/chronic_kidney_disease.nightly.spec.ts:19]: I think you highlighted the removed side of the diff here, right? I should have removed all these lines as part of the PR

**Review by kccrtv — APPROVED** [review]
😎 dope! thanks ben!

**Pull request merged**